lib/debugger/debugger.dart

```diff
@@ -0,0 +1,1309 @@
+
+      StreamSubscription sub;
+
+      sub = element.element.onMouseOver.listen((e) {
+        // TODO(devoncarew): Call toString() only after a short dwell.
+        sub.cancel();
+        variableDescriber(variable).then((String desc) {
+          element.tooltip = desc;
+        });
+      });
+
+      return element;
+    });
+  }
+
+  SelectableList<BoundVariable> _items;
+
+  List<BoundVariable> get items => _items.items;
+
+  CoreElement get element => _items;
+
+  void showVariables(Frame frame) {
+    // AsyncCausal frames don't have local vars.
+    _items.setItems(frame.vars ?? <BoundVariable>[]);
+  }
+
+  void clearVariables() {
+    _items.setItems(<BoundVariable>[]);
+  }
+}
+
+class BreakOnExceptionControl extends CoreElement {
+  BreakOnExceptionControl() : super('div', classes: 'break-on-exceptions') {
+    final CoreElement unhandled = new CoreElement('input')
+      ..setAttribute('type', 'checkbox');
+    _unhandledElement = unhandled.element;
+
+    final CoreElement all = new CoreElement('input')
+      ..setAttribute('type', 'checkbox');
+    _allElement = all.element;
+
+    add([
+      span(text: 'Break on: '),
+      new CoreElement('label')
+        ..add(<CoreElement>[unhandled, span(text: ' Unhandled exceptions')]),
+      new CoreElement('label')
+        ..add(<CoreElement>[all, span(text: ' All exceptions')]),
+    ]);
+
+    unhandled.element.onChange.listen((_) {
+      _pauseModeController.add(exceptionPauseMode);
+    });
+
+    all.element.onChange.listen((_) {
+      if (_allElement.checked) {
+        unhandled.enabled = false;
+        _unhandledElement.checked = true;
+      } else {
+        unhandled.enabled = true;
+      }
+      _pauseModeController.add(exceptionPauseMode);
+    });
+  }
+
+  html.InputElement _unhandledElement;
+  html.InputElement _allElement;
+
+  final StreamController<String> _pauseModeController =
+      new StreamController.broadcast();
+
+  /// See the string values for [ExceptionPauseMode].
+  Stream<String> get onPauseModeChanged => _pauseModeController.stream;
+
+  String get exceptionPauseMode {
+    if (_allElement.checked) {
+      return ExceptionPauseMode.kAll;
+    } else if (_unhandledElement.checked) {
+      return ExceptionPauseMode.kUnhandled;
+    } else {
+      return ExceptionPauseMode.kNone;
+    }
+  }
+
+  set exceptionPauseMode(final String value) {
+    if (value == ExceptionPauseMode.kAll) {
+      _allElement.checked = true;
+      _unhandledElement.checked = true;
+      _unhandledElement.setAttribute('disabled', '');
+    } else if (value == ExceptionPauseMode.kUnhandled) {
+      _allElement.checked = false;
+      _unhandledElement.checked = true;
+      _unhandledElement.attributes.remove('disabled');
+    } else {
+      _allElement.checked = false;
+      _unhandledElement.checked = false;
+      _unhandledElement.attributes.remove('disabled');
+    }
+  }
+}
+
+class ScriptAndPos {
+  ScriptAndPos(this.uri, {this.position});
```

Comment:
nit: add @required to position argument assuming it is required.

Reply:
done